test/elixir_bench_web/schema/schema_test.exs

```diff
@@ -79,30 +81,403 @@ defmodule ElixirBenchWeb.SchemaTest do
+              "commit" => %{
+                "message" => "#{job.commit_message}",
+                "sha" => "#{job.commit_sha}",
+                "url" => "#{job.commit_url}"
+              },
+              "environment" => %{
+                "cpu" => "#{job.cpu}",
+                "cpuCount" => job.cpu_count,
+                "elixirVersion" => "#{job.elixir_version}",
+                "erlangVersion" => "#{job.erlang_version}",
+                "memory" => job.memory_mb
+              },
+              "result" => %{
+                "average" => measurement.average,
+                "ips" => measurement.ips,
+                "maximum" => measurement.maximum,
+                "median" => measurement.median,
+                "minimum" => measurement.minimum,
+                "mode" => measurement.mode,
+                "sampleSize" => measurement.sample_size,
+                "stdDev" => measurement.std_dev,
+                "stdDevIps" => measurement.std_dev_ips,
+                "stdDevRatio" => measurement.std_dev_ratio
+              }
+            }
+          }
+        }
+
+      assert ^json_data = resp
+    end
+
+    test "return not found message when measurement does not exist", context do
+      query = """
+        measurement (id: 9999)
+        {
+          id
+          collectedAt
+        }
+      """
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      error_message = Enum.at(resp["errors"], 0)
+
+      assert %{"message" => "not_found"} = error_message
+    end
+
+    test "return required field message when missing fields", context do
+      required_fields = ["id"]
+
+      query = """
+        measurement {
+          id
+        }
+      """
+
+      resp_body = context.conn
+        |> execute_query(query)
+        |> Map.get(:resp_body)
+
+      Enum.each(required_fields, fn field ->
+        assert(
+          Regex.match?(~r/errors.*#{field}.*found null/, resp_body),
+          "Field '#{field}' not found in #{resp_body}"
+        )
+      end)
+
+    end
+  end
+
+  describe "jobs query" do
+    test "list all jobs", context do
+      jobs_ids =
+      insert_list(3, :job)
+      |> Enum.map(fn job -> Integer.to_string(job.id) end)
+
+      query = """
+        jobs {
+          id
+        }
+      """
+
+      json_data =
+      %{
+        "data" => %{
+           "jobs" => [
+             %{"id" => Enum.at(jobs_ids, 0)},
+             %{"id" => Enum.at(jobs_ids, 1)},
+             %{"id" => Enum.at(jobs_ids, 2)}
+           ]
+         }
+       }
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      assert ^json_data = resp
+    end
+
+    test "return empty list when there is no registered jobs", context do
+      query = """
+        jobs {
+          id
+        }
+      """
+
+      json_data =
+      %{
+        "data" => %{
+           "jobs" => []
+         }
+       }
```

Comment:
as we have the empty case quite some times and it's always the same we could put the assertion and the empty data in a helper function or we could also define the empty json as a module attribute and just reuse it :)